### PR TITLE
Expose Porter domains using helper methods

### LINKED ISSUE #311: Update Porter network configuration for `@nucypher/*` packages
- Requested in https://github.com/nucypher/nucypher-ts/pull/301#discussion_r1347166596
- Create separate network configs for `@nucypher/taco` and `@nucypher/pre`

### PR DESCRIPTION
**Pull request by piotr-roslaniec**
**Type of PR:**
- Feature

**Required reviews:** 
- 1

**Issues fixed/closed:**
- Fixes #311

**Notes for reviewers:**
- Based on #301 
- In hindsight, could have been part of #301
- Rebased over https://github.com/nucypher/nucypher-ts/pull/315
- [ ] Support `tapir`, `mainnet` and `lynx` networks. Test examples and demos against lynx.

### CODE REVIEW

File: packages/pre/src/porter.ts [hunk 1]
```diff
@@ -0,0 +1,17 @@
+import { getPorterUri as doGetPorterUri } from '@nucypher/shared';
+
+type PreDomain = 'oryx';
+
+export const PRE_DOMAIN: Record<string, PreDomain> = {
+  TESTNET: 'oryx',
```
Comment (derekpierre) [inline packages/pre/src/porter.ts:6]: Was this supposed to be `tapir`?

```suggestion
  TESTNET: 'tapir',
```

File: packages/shared/src/porter.ts [hunk 2]
```diff
@@ -12,21 +12,20 @@ import qs from 'qs';
...
   mainnet: 'https://porter.nucypher.community',
   tapir: 'https://porter-tapir.nucypher.community',
   oryx: 'https://porter-oryx.nucypher.community',
   lynx: 'https://porter-lynx.nucypher.community',
 };
+export type PorterNetwork = keyof typeof porterUri;
 
-export const getPorterUri = (network: Network): string => {
-  const uri = PORTER_URIS[network];
+export const getPorterUri = (network: PorterNetwork): string => {
```
Comment (derekpierre) [inline packages/shared/src/porter.ts:23]: Can we use "domain" instead of "network"?

File: packages/pre/src/porter.ts [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import { getPorterUri as doGetPorterUri } from '@nucypher/shared';
+
+type PreDomain = 'oryx';
```
Comment (KPrasch) [inline packages/pre/src/porter.ts:3]: This domain is planned to be EOL after the next release, Shall we update it? 

```suggestion
type PreDomain = PRE_DOMAIN.TESTNET;
```

Comment (piotr-roslaniec) [reply packages/pre/src/porter.ts:3]: We can update it now, sure. I gather from comments in this PR that it's supposed to be `tapir`?

WRT to the line: Notice how `type PreDomain = 'oryx';` is used in `PRE_DOMAIN` type definition:

```
export const PRE_DOMAIN: Record<string, PreDomain> = {
  TESTNET: 'oryx',
};
```

It works here like an enum - It describes the legal values in `PRE_DOMAIN` dictionary.

File: packages/pre/src/porter.ts (hunk 1 above)
Comment (piotr-roslaniec) [reply packages/pre/src/porter.ts:6]: Support `tapir`, `mainnet` and `lynx` networks. Test examples and demos against `lynx`.

Comment (piotr-roslaniec) [reply packages/pre/src/porter.ts:6]: @derekpierre What was the conclusion here? To support `tapir`, `mainnet`, and `lynx` networks in both PRE and TACo?

Comment (derekpierre) [reply packages/pre/src/porter.ts:6]: Yes. All possible TACo domains should be allowed: `tapir`, `lynx`, `mainnet`.

**Review by KPrasch — APPROVED** [review]

**Pull request merged**